src/types/workshopContext.ts

```diff
@@ -202,3 +202,16 @@ export type IWorkshopContextField<
   T extends IConfigDefinition,
   V extends IVariableType_WithDefaultValue
 > = ValueAndSetterMethods<V> | ExecutableEvent | IWorkshopContext<T>[];
+
+/**
+ * Extends the IWorkshopContext with a setHeight function
+ */
+export type IWorkshopContextWithHeight<T extends IConfigDefinition> = IWorkshopContext<T> & {
+  /**
+   * Sets the maximum height of the iframe in Workshop.
+   * Only has an effect when the app is running inside an iframe AND
+   * the Workshop widget's height is set to "auto (max)".
+   * @param height The maximum height in pixels
+   */
+  setAutoMaxHeight: (height: number) => void;
```

Comment:
I wonder if there would be an issue if one of the fields in the config definition were to be named `setAutoMaxHeight`... and what would happen in that case of conflict. Perhaps for this we should have `IWorkshopContextWithHeight` be `{ context: IWorskhopContext<T>; setAutoMaxHeight: setAutoMaxHeight: (height: number) => void;}`

Reply:
But I think if we do this we can't break existing usage so we should like pass a flag and change the interface to calling `useWorshopContext` to like 
```
useWorkshopContext(CONFIG, { enableSetAutoMaxHeight: true })
```
if the option enableSetAutoMaxHeight is not given or false, return  `IWorkshopContext`, but if it's true return `IWorkshopContextWithHeight`